tests/server/onLoadDocument.ts

```diff
@@ -164,60 +216,6 @@ test('stops when an error is thrown in onLoadDocument', async t => {
   })
 })
 
-test('disconnects all clients related to the document when an error is thrown in onLoadDocument', async t => {
-  const resolvesNeeded = 4
-
-  await new Promise(async resolve => {
-
-    const server = await newHocuspocus({
-      async onLoadDocument() {
-        return new Promise((resolve, fail) => {
-          setTimeout(() => {
-            // eslint-disable-next-line prefer-promise-reject-errors
```

Comment:
Why remove this test?

Reply:
This test is removed because this behavior should not exist anymore.  This test is asserting that if the `onLoadDocument` throws then `provider2` should still be able to connect to it, which is what we do not want.

Reply:
But the error may be transient, so if it throws it should be able to retry, no?

Reply:
I'm not quite sure i follow.  Wouldn't it be up to the `HocuspocusProvider` to handle retry logic once the connection fails to be established.

Reply:
If I add the following console logs to the test i see:

```
test.only('disconnects all clients related to the document when an error is thrown in onLoadDocument', async t => {
  const resolvesNeeded = 4

  await new Promise(async resolve => {

    const server = await newHocuspocus({
      async onLoadDocument() {
        return new Promise((resolve, fail) => {
          setTimeout(() => {
            // eslint-disable-next-line prefer-promise-reject-errors
            fail('ERROR')
          }, 250)
        })
      },
      async onStoreDocument(data) {
        t.fail('MUST NOT call onStoreDocument')
      },
    })

    let resolvedNumber = 0
    const resolver = () => {
      resolvedNumber += 1

      if (resolvedNumber >= resolvesNeeded) {
        t.is(server.documents.size, 0)
        t.is(server.getConnectionsCount(), 0)
        resolve('done')
      }
    }

    const provider1 = newHocuspocusProvider(server, {
      onConnect() {
        console.log('provider1 connect')
        resolver()
      },
      onClose(event) {
        provider1.disconnect()
        console.log('provider1 disconnect')
        resolver()
      },
    })

    const provider2 = newHocuspocusProvider(server, {
      onConnect() {
        console.log('provider2 connect')
        resolver()
      },
      onClose() {
        provider2.disconnect()
        console.log('provider2 disconnect')
        resolver()
      },
    })

  })

})
```


```
provider2 connect
provider1 disconnect
provider2 disconnect
provider2 disconnect
```

I feel like this is not what i would expect this behavior to do in the first place.